Modules/System/Geolocation/src/GeolocationImpl.Codeunit.al

```diff
@@ -0,0 +1,141 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+codeunit 50101 "Geolocation Impl."
+{
+    Access = Internal;
+
+    var
+        GeolocationPage: Page Geolocation;
+        LocationNotRetrievedError: Label 'The geographical location data was not retrieved.';
+        CachedLocation: Dotnet Location;
+        LocationProvider: DotNet LocationProvider;
+        LocationOptions: DotNet LocationOptions;
+        LocationOptionsEnabled: Boolean;
+
+    procedure RequestGeolocation(GeolocationImpl: Codeunit "Geolocation Impl."): Boolean
+    begin
+        Clear(GeolocationPage);
+        GeolocationPage.SetGeolocationImpl(GeolocationImpl);
+        GeolocationPage.RunModal();
+        if HasGeolocation() then
+            exit(true);
+
+        exit(false);
+    end;
+
+    procedure LocationInteractionOnOpenPage(var LocationProvdr: Dotnet LocationProvider; var LocationAvailable: Boolean)
```

Comment:
Why not hide `LocationProvdr` behind an interface and provide a mock implementation in the tests?

@darjoo, what do you think?

Reply:
Since the DotNet has to be on the page due to the page trigger, It would only be used in GeoLocationImpl. And the mocking would only be specifically for one function, IsAvailable.

I'm inclined to say not to move it behind an interface. He has also included a event already to mock the value for that test.